doc/services/portability/posix/option_groups/posix-c-lang-support.rst

```diff
@@ -0,0 +1,127 @@
+.. _posix_option_groups_external:
+
+External Subprofiling Option Groups
+###################################
+
+.. _posix_option_group_c_lang_support:
+
+POSIX_C_LANG_SUPPORT
+====================
+
+The ``POSIX_C_LANG_SUPPORT`` option group is supplied by the ISO C Library.
+
+Zephyr supports most ISO C features pertaining to the OS. These features are cooperatively
+maintained by the C Library and POSIX API maintainers.
+
+For more information on developing Zephry applications in C, or other programming languages,
+please refer to :ref:`details<language_support>`.
+
+.. csv-table:: POSIX_C_LANG_SUPPORT
```

Comment:
What's the rationale for the functions listed as "no" below? Like, `asctime()` is available in Newline and Picolibc, and even used in the gettimeofday sample?

Reply:
That's a good question. I assumed the C Library maintainer would be updating this list periodically as well. Hence, the PR.

Reply:
I guess we can only declare functions as supported if they are also supported in the minimal libc, and it looks like `asctime()` is unavailable there. I would guess that's the case for most other unimplemented functions.

However, picolibc, and newlib can supply most (if not all) of these functions, and picolibc is IIRC the default now (but only if either the Zephyr SDK is available or picolibc is available as a module).

Reply:
Added some paragraphs explaining the significance of this table.

Reply:
> I honestly wonder if maybe this list should be moved to [Minimal libc](https://docs.zephyrproject.org/latest/develop/languages/c/minimal_libc.html) since it appears to mainly reflect completion status of that library.

+1. Keeping the table is making the whole page unnecessarily verbose IMO, i.e. I don't think it's the right place for delving into Zephyr history, and it should rather focus on current/default behavior. In short: I would remove the table altogether, and remove L14-L29 to replace it with something along the lines of the second note ("When using Newlib, Picolibc (default C library as of Zephyr 3.5), or other C libraries conforming to the ISO C Standard, the entire
   ``POSIX_C_LANG_SUPPORT`` Option Group is considered supported.

Reply:
Docs are building fine on my end now - I updated to include a couple of things that I was surprised were missing. 

`POSIX_C_LANG_JUMP` (already supported, + link to details)
`POSIX_C_LANG_MATH` (+ link to details)

Also made a small comment that should clarify that the C library is also a perfectly valid source for conforment implementations of POSIX API (of course). It depends on the C library and the implementation, as to whether it's going to work with Zephyr, of course. E.g. should pass testsuite.

Reply:
@kartben , @keith-packard - PTAL